### ISSUE TITLE
can't migrate as long as there are compile errors

### ISSUE THREAD

**Issue by axaluss** [flyway/flyway#979]
I try to flywayMigrate but there are compile errors. 
So the migrate task is never executed.

The funny thing is, that i can only generate the proper code with jooq **after** i migrated the database.

Is there a way get around the compile dependency?

**Comment by axaluss** [comment]
have a look at this fix: https://github.com/axaluss/FlywaySbt

**Comment by axelfontaine** [comment]
Thanks for the report. This is something I plan to change for all build tool clients (ant, maven, gradle, sbt) in time for 4.0

**Comment by nkgm** [comment]
+1
On maven a workaround is possible. SBT however requires more manual labor.

**Comment by axelfontaine** [comment]
The default location for Maven, Gradle and SBT has been change to filesystem:src/main/resources/db/migration which should improve integration with jOOQ and remove the need for compilation when only using SQL files

**Comment by nafg** [comment]
The fix breaks sbt subprojects (I had to add `flywayLocations := Seq("db/migration")`) since they are in a different subdirectory. I guess it would also not work with Play's directory layout or any other customization.

**Comment by mikebridge** [comment]
Is there an example of how to set this up in sbt?  I still have a dependency on compilation in a subproject with 4.0.3:

```scala
lazy val `my-subproject` = (project in file("my-subproject"))
 .settings(
    flywayUrl := "jdbc:postgresql://localhost:5432/mydb",
    flywayLocations := Seq("db/migration"),
    flywayUser := "myuser")
```

**Comment by nafg** [comment]
@mikebridge what does `inspect tree flywayMigrate` say the dependency is?

Also why do you need to override `flywayLocations`?

**Comment by mikebridge** [comment]
@nafg It looks like flywayMigrate depends on fullClasspath.

As for `flywayLocations`,  I believe I applied your May 30 workaround for subprojects

```
> inspect tree identity-manager/flywayMigrate
[info] identity-manager/*:flywayMigrate = Task[Unit]
[info]   +-identity-manager/*:flywayConfig = Task[org.flywaydb.sbt.FlywayPlugin$Config]
[info]   +-identity-manager/runtime:fullClasspath = Task[scala.collection.Seq[sbt.Attributed[java.io.File]]]
[info]     +-identity-manager/runtime:exportedProducts = Task[scala.collection.Seq[sbt.Attributed[java.io.Fil..
[info]     | +-identity-manager/compile:packageBin::artifact = Artifact(identity-manager,jar,jar,None,List(co..
[info]     | +-identity-manager/compile:packageBin::artifactPath = identity-manager/target/scala-2.11/identit..
[info]     | +-identity-manager/runtime:productDirectories = Task[scala.collection.Seq[java.io.File]]
[info]     | | +-identity-manager/compile:classDirectory = identity-manager/target/scala-2.11/classes
```

**Comment by nafg** [comment]
Hmm, I suspect it might not be necessary on the latest version but I'm not sure

**Comment by warpian** [comment]
the fix for sbt in flyway 4.2 is not working in my project. the cyclic dep is still there.
